### DISCUSSION TITLE
Set focus on newly inserted input element

### DISCUSSION THREAD

**Discussion by cfprotools** [livewire/livewire#2929] [1. Help]
I've got an app I'm building that is basically an outline building tool.  We've got modules and notes that each have "titles".  I want the user to be able to hit enter to insert a new note into a module and then immediately focus on the new input so they can start typing.  I've got the new note being created and inserted and am firing an event to set the focus to the new element, but the event fires before the refresh so the input element doesn't exist in the DOM yet when the Alpine event fires to set the focus.  I've had to workaround it by setting a 500ms delay on firing the event to set the focus, but this makes the UI feel sluggish.  I'd love for the focus to be set automatically as soon as the input is inserted into the DOM.

Anyone have any examples of something similar that I could dissect?  Any ideas on how to accomplish this with Livewire and/or Alpine?

**Comment by PR4M** [comment]
I don't read your description fully, sorry, 

but I just found something that might worth to check??

[Set focus on input element with Alpine js](https://egghead.io/lessons/javascript-focus-an-input-field-on-button-click-with-x-ref-and-the-refs-property-in-alpine-js)

**Reply by jeffchown** [reply]
@cfprotools Have a look at $nextTick in Alpine (https://alpinejs.dev/magics/nextTick)
